langchain/prompts/base.py

```diff
@@ -32,11 +32,35 @@ def check_valid_template(
         raise ValueError("Invalid prompt schema.")
 
 
+class BaseOutputParser(ABC):
+    """Class to parse the output of an LLM call."""
+
+    @abstractmethod
+    def parse(self, text: str) -> Union[str, List[str], Dict[str, str]]:
+        """Parse the output of an LLM call."""
+
+
+class ListOutputParser(ABC):
```

Comment:
I honestly might prefer parser to be a bit less defined -- e.g. not sure there is a need for a ListOutputParser primitive -- feels like the user could fully define their own OutputParser

Reply:
oh they def can! but some chains (like the for loop chain) will expect a list output...